plugins/events-backend-module-kafka/src/publisher/config.ts

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Config, readDurationFromConfig } from '@backstage/config';
+import { ConsumerConfig, ConsumerSubscribeTopics, KafkaConfig } from 'kafkajs';
+import { Duration } from 'luxon';
+
+/**
+ * @public
+ */
+export interface KafkaConsumerConfig {
+  backstageTopic: string;
+  consumerConfig: ConsumerConfig;
+  consumerSubscribeTopics: ConsumerSubscribeTopics;
+}
+
+/**
+ * @public
+ */
+export interface KafkaEventSourceConfig {
+  kafkaConfig: KafkaConfig;
+  kafkaConsumerConfigs: KafkaConsumerConfig[];
+}
+
+const CONFIG_PREFIX_PUBLISHER =
+  'events.modules.kafka.kafkaConsumingEventPublisher';
+
+/**
+ * Reads an optional HumanDuration from the config and returns the value in milliseconds if the key is defined.
+ *
+ * @param config - The configuration object to read from.
+ * @param key - The key to look up in the configuration.
+ * @returns The duration in milliseconds, or undefined if the key is not defined.
+ */
+const readOptionalHumanDurationInMs = (
+  config: Config,
+  key: string,
+): number | undefined => {
+  const humanDuration = config.has(key)
+    ? readDurationFromConfig(config, { key })
+    : undefined;
+
+  if (!humanDuration) return undefined;
+
+  return Duration.fromObject(humanDuration).as('milliseconds');
```

Comment:
There's a widely used `durationToMilliseconds` in `@backstage/types` if you want to get rid of the luxon dependency.